src/test/LockedWithdrawal.t.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {TestPlus} from "src/test/TestPlus.sol";
+import {Deploy} from "./Deploy.sol";
+
+import {BaseVault} from "src/vaults/cross-chain-vault/BaseVault.sol";
+import {AffineVault} from "src/vaults/AffineVault.sol";
+import {LockedWithdrawalEscrow} from "src/vaults/LockedWithdrawal.sol";
+
+contract LockedWithdrawalTest is TestPlus {
+    BaseVault vault;
+
+    LockedWithdrawalEscrow withdrawalEscrow;
+    uint256 sla;
+
+    function setUp() public {
+        sla = 10;
+        vault = Deploy.deployL2Vault();
+        withdrawalEscrow = new LockedWithdrawalEscrow(AffineVault(address(vault)), sla);
+    }
+
+    function testRegisterForWithdrawal() public {
+        vm.prank(address(vault));
+        withdrawalEscrow.registerWithdrawalRequest(alice, 1000);
+
+        assertEq(withdrawalEscrow.balanceOf(alice), 1000);
+
+        assertEq(withdrawalEscrow.canWithdraw(alice), false);
+
+        //withdrawable amount is zero due to no funds as asset
+        assertEq(withdrawalEscrow.withdrawableAmount(alice), 0);
+    }
+
+    function testResolvePendingDebt() public {
+        // do a withdrawal request
+        vm.warp(1_641_070_800);
```

Comment:
I think it would be clearer if you used a `startTime` variable here. Then in the second warp below you could use `vm.warp(startTime + sla)`